### PR TITLE
Add check to keep releases.json in sync with the firmware binaries

### PR DESCRIPTION
**Pull request by peter-sanderson**
This PR adds check to validate presence of the binaries as declared in the `releases.json`

If some binary is missing:
image

If there is binary but is missing from `releases.json`
image

It also adds a `shellcheck`

### CODE REVIEW

File: ci/s3sync.sh [hunk 1]
```diff
@@ -6,11 +6,9 @@
 
 DIRS="bootloader bridge firmware legal registry udev suite connect security transparency misc"
 BUCKET=data.trezor.io
-ROLLBACK=rollback-data.trezor.io
```
Comment (peter-sanderson) [inline ci/s3sync.sh]: Unused (`shellcheck` told me)

File: firmware/t3b1/releases.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (peter-sanderson) [inline firmware/t3b1/releases.json]: Is this safe to add empty? Its just for consistency.

File: ci/s3sync.sh
Comment (peter-sanderson) [inline ci/s3sync.sh:1]: It may be worth to unify `ci` and `scripts` folders? @vdovhanych WDYT?

File: firmware/t3b1/releases.json (hunk 2 above)
Comment (komret) [reply firmware/t3b1/releases.json]: Should be safe.

File: scripts/run-releases-json-for-all-devices.sh
Comment (komret) [inline scripts/run-releases-json-for-all-devices.sh:1]: Could this be included in the `check-firmware-presence-in-releases-json.sh` file, calling `compare_files ` in the loop?

**Review by komret — DISMISSED** [review]
The shellcheck addition could have been a separate commit.

File: scripts/run-releases-json-for-all-devices.sh
Comment (peter-sanderson) [reply scripts/run-releases-json-for-all-devices.sh:1]: I liked I can just run it one-by one when using it locally. But this is ok as well WDYT?

Comment (komret) [reply scripts/run-releases-json-for-all-devices.sh:1]: OK

File: .github/workflows/check-shell-validation.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: "[Check]: Shell validation"
```
Comment (vdovhanych) [inline .github/workflows/check-shell-validation.yml]: remove `:`

File: ci/s3sync.sh
Comment (vdovhanych) [reply ci/s3sync.sh:1]: yeah, we can move the release script into the scripts folder, but it's not necessary, tbh.

**Review by vdovhanych — CHANGES_REQUESTED** [review]
Adding shellcheck validation should be a separate commit and please fix one small nitpick.

File: scripts/check-firmware-presence-in-releases-json.sh [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+trezor-t1b1-1.0.0.bin
+trezor-t1b1-1.1.0.bin
+trezor-t1b1-1.2.0.bin
+trezor-t1b1-1.2.1.bin
+trezor-t1b1-1.3.0.bin
+trezor-t1b1-1.3.1.bin
+trezor-t1b1-1.3.2.bin
+trezor-t1b1-1.3.3.bin
+trezor-t1b1-1.3.4.bin
+trezor-t1b1-1.3.5.bin
```
Comment (peter-sanderson) [inline scripts/check-firmware-presence-in-releases-json.sh]: @prusnak We are not having those old FW binary files. Is it ok to just ignore them from this check?

We have them in `releases.json` so this check expects them in the directory, but they are not here.

Comment (szymonlesisz) [reply scripts/check-firmware-presence-in-releases-json.sh]: i think we should remove it from `releases.json` as they are not supported (<1.9.0)

Comment (peter-sanderson) [reply scripts/check-firmware-presence-in-releases-json.sh]: OK, did it here: https://github.com/trezor/data/pull/90/commits/b8000706a8c58bb2518052b3456eca5b87c2e8a4

**Comment by peter-sanderson** [comment]
I have also just found that we have https://github.com/trezor/data/blob/master/check_releases.py

It seems to be checking the existence of binaries for latest release + also checking presence of an `changelog` , some header checking.


However it checks only `Trezor One` and `Trezor T` :exclamation: . We shall consider unifying those scripts and making it auto-magically work for all devices (when we add new one).

**Comment by vdovhanych** [comment]
> I have also just found that we have [`master`/check_releases.py](https://github.com/trezor/data/blob/master/check_releases.py?rgh-link-date=2024-08-09T09%3A39%3A56Z)
> 
> It seems to be checking the existence of binaries for latest release + also checking presence of an `changelog` , some header checking.
> 
> However it checks only `Trezor One` and `Trezor T` ❗ . We shall consider unifying those scripts and making it auto-magically work for all devices (when we add new one).

Yes, we have such a script, but that is not true. It is checking t1b1, t2t1, and t2b1. t3t1 is missing from there. I'm not sure how you would make it work automagically, but if you have an idea, go for it.

**Review by komret — DISMISSED** [review]
I'd rather see this integrated with `check_releases.py`, but now that it is done like this, let's use it. @vdovhanych, can you approve?

**Review by vdovhanych — DISMISSED** [review]
i guess its fine like this for now

File: firmware/2/releases.json [hunk 5]
```diff
@@ -8,10 +8,10 @@
         "bootloader_version": [2, 1, 6],
         "firmware_revision": "632b9561559b7ab6824bb7eeac072874e07b7b82",
         "translations": ["cs-CZ", "de-DE", "es-ES", "fr-FR"],
-        "url": "data/firmware/t2t1/trezor-t2t1-2.8.1.bin",
+        "url": "data/firmware/2/trezor-2.8.1.bin",
         "fingerprint": "d3af84a212d32785449ca6575e3cf2a641920b353a82dec9f059083ea5d4b149",
         "changelog": "* Fixed displaying of a progress indicator for the formatting operation.\n* Improve precision of PIN validation countdown.\n* Solana: Improved support for AToken Create operation.",
-        "url_bitcoinonly": "data/firmware/t2t1/trezor-t2t1-2.8.1-bitcoinonly.bin",
+        "url_bitcoinonly": "data/firmware/2/trezor-2.8.1-bitcoinonly.bin",
```
Comment (peter-sanderson) [inline firmware/2/releases.json]: @matejcik :pray: is this ok? Or did you done this deliberately?

**Review by Lemonexe — APPROVED** [review]
LGTM :rocket:

**Review by vdovhanych — APPROVED** [review]

**Pull request merged**